dashboard/src/components/Directory/Map.vue

```diff
@@ -1,30 +1,215 @@
 <template>
   <l-map style="height: 100%" :zoom="zoom" :center="center">
     <l-tile-layer :url="url" :attribution="attribution"></l-tile-layer>
-    <l-marker :lat-lng="markerLatLng"></l-marker>
+    <l-marker :lat-lng="markerLatLng" @click="getEvent">
```

Comment:
with this behavior we need have a modal for each marker, I suggest a method that get data, fill the pop up, and show the popo up, and the pop up can be a component